profiles/fscloud/variables.tf

```diff
@@ -0,0 +1,206 @@
+  type        = string
+  default     = null
+}
+
+variable "cos_plan" {
+  description = "Plan to be used for creating cloud object storage instance. Only used if 'create_cos_instance' it true."
+  type        = string
+  default     = "standard"
+  validation {
+    condition     = contains(["standard", "lite"], var.cos_plan)
+    error_message = "The specified cos_plan is not a valid selection!"
+  }
+}
+
+##############################################################################
+# COS bucket variables
+##############################################################################
+
+variable "create_cos_bucket" {
+  description = "Set as true to create a new Cloud Object Storage bucket"
+  type        = bool
+  default     = true
+}
+
+variable "primary_region" {
+  description = "region for the primary bucket"
+  type        = string
+  default     = "us-south"
+}
+
+variable "secondary_region" {
+  description = "region for the secondary bucket"
+  type        = string
+  default     = "us-east"
+}
+
+variable "bucket_name" {
+  type        = string
+  description = "The name to give the newly provisioned COS bucket. Only required if 'create_cos_bucket' is true."
+  default     = null
+}
+
+variable "bucket_storage_class" {
+  type        = string
+  description = "the storage class of the newly provisioned COS bucket. Only required if 'create_cos_bucket' is true. Supported values are 'standard', 'vault', 'cold', and 'smart'."
+  default     = "standard"
+
+  validation {
+    condition     = can(regex("^standard$|^vault$|^cold$|^smart$", var.bucket_storage_class))
+    error_message = "Variable 'bucket_storage_class' must be 'standard', 'vault', 'cold', or 'smart'."
+  }
+}
+
+variable "activity_tracker_crn" {
+  type        = string
+  description = "Activity tracker crn for COS bucket (Optional)"
+  default     = null
+}
+
+variable "sysdig_crn" {
+  type        = string
+  description = "Sysdig Monitoring crn for COS bucket (Optional)"
+  default     = null
+}
+
+##############################################################################
+# COS bucket encryption variables
+##############################################################################
+
+variable "primary_existing_hpcs_instance_guid" {
+  description = "The GUID of the Hyper Protect Crypto service in which the key specified in var.hpcs_key_crn is coming from. Required if var.create_cos_instance is true in order to create an IAM Access Policy to allow Key protect to access the newly created COS instance."
+  type        = string
+}
+
+variable "secondary_existing_hpcs_instance_guid" {
+  description = "The GUID of the Hyper Protect Crypto service in which the key specified in var.hpcs_key_crn is coming from. Required if var.create_cos_instance is true in order to create an IAM Access Policy to allow Key protect to access the newly created COS instance."
+  type        = string
+}
```

Comment:
should we have some kind of variable validation to ensure that the GUID passed for these is indeed a hyper protect GUID and not a Key Protect one? Because I think it would allow a key protect guid to be passed in, but it would fail on the auth policy creation. however auth policy creation is optional so it may not fail at all? Perhaps validation that the GUID contains the string value contains `.*hs-crypto.*` ?

Reply:
For this one, we allow KP for demo purposes. Otherwise, the example cannot be run externally unless Hyper protect instances are provided. This is stated in the example readme. If we want to force only Hyper protect then we can

Reply:
If you want to support key protect, then won't the auth policy code need to be updated, as its locked into `target_service_name = "hs-crypto"`

If we allow key protect, won't the profile no longer be fscloud compliant?

Reply:
I forgot about the policy changes that where made. I will make it HPCS only